### PR TITLE
Updated the structure and applied feedback

### PR DESCRIPTION
**Pull request by Felonious-Spellfire**
## What does this PR do?

Modifies the content for Encrypted Secrets.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [X] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by alecmerdler — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by edenlightning — COMMENTED** [review]
We should also add this to the menu in how to "Add a secret"

File: docs/source-app/glossary/secrets.rst [hunk 1]
```diff
@@ -4,54 +4,99 @@
...
 
 .. tip::
 	For non-sensitive configuration values, we recommend using :ref:`plain-text Environment Variables <environment_variables>`.
 
-*******************
-Overview of Secrets
-*******************
-
-The ``--secret`` option has been added to the **lightning run app** command. ``--secret`` can be used by itself or alongside ``--env``.
+***************
```
[the remark this answers is not in the document]
Comment (Felonious-Spellfire) [reply docs/source-app/glossary/secrets.rst:12]: This page adheres to the guideline. Everything on the checklist is in this topic. I'll modify the video. I think it's important that we are able to effectively articulate to users what we did and the reason why we did it. This helps our own understanding of the feature and that of the users as well. I can make a second topic that is purely action-oriented, (Adding Secrets), but this topic will remain in the structure as it is for the glossary.

File: docs/source-app/glossary/secrets.rst [hunk 2]
```diff
@@ -4,54 +4,99 @@
...
 
+**********************
+What were we thinking?
+**********************
+
+Many Apps require access to private data like API keys, access tokens, database passwords, or other credentials. You need to protect this data.
+We developed this feature to provide you with a secure way to store this data in a way that is accessible to Apps so that they can authenticate third-party services/solutions.
+
+----
+
```
[the remark this answers is not in the document]
Comment (Felonious-Spellfire) [reply docs/source-app/glossary/secrets.rst:30]: See my comment above

File: docs/source-app/glossary/secrets.rst [hunk 3]
```diff
@@ -4,54 +4,99 @@
...
 *********************
 Use Encrypted Secrets
 *********************
 
-First, a Secret must be created using the admin web UI. Once you create a Secret, you can bind it to any of your Apps. You do not need to create a new Secret for each App if the Secret value is the same.
+To use Encrypted Secrets:
 
-.. note::
-    Secret names must start with a letter and can only contain letters, numbers, dashes, and periods. The Secret names must comply with `RFC1123 naming conventions <https://www.rfc-editor.org/rfc/rfc1123>`_. The Secret value has no restrictions.
+#. Log in to your lightning.ai account, go to **Secrets**, and create the Secret (provide a name and value for the secret).
```
[the remark this answers is not in the document]
Comment (Felonious-Spellfire) [reply docs/source-app/glossary/secrets.rst:37]: I'll add a video. Throughout all of the other docs I don't see the action, feedback, action, step by step sort of content. 

If we want to move to something like that, then lets have a discussion about it.

File: docs/source-app/glossary/secrets.rst [hunk 4]
```diff
@@ -4,54 +4,99 @@
...
+
+    .. note:: Once you create a Secret, you can bind it to any of your Apps. You do not need to create a new Secret for each App if the Secret value is the same.
```
[the remark this answers is not in the document]
Comment (Felonious-Spellfire) [reply docs/source-app/glossary/secrets.rst:39]: I'll remove the first part, the second sentence stays as a note.

**Comment by krishnakalyan3** [comment]
@alecmerdler @nohalon  should we mention in the doc somewhere that `_` are not supported?

**Comment by Felonious-Spellfire** [comment]
> @alecmerdler @nohalon should we mention in the doc somewhere that `_` are not supported?

We do:

Secret names must start with a letter and can only contain letters, numbers, dashes, and periods.

**Pull request merged**